commit e7e4f81de27bceef982eac89b56c9d28e89403e5 (refs/changes/62/662/1)
Author: Alex Wilson <alex.wilson@joyent.com>
Date:   2016-10-12T18:23:37-07:00 (3 years ago)
    
    joyent/node-cueball#39 ConnectionFSM#state_idle calls S.on after S.gotoState

diff --git a/lib/connection-fsm.js b/lib/connection-fsm.js
index 395a21d..8cdacb9 100644
--- a/lib/connection-fsm.js
+++ b/lib/connection-fsm.js
@@ -393,8 +393,8 @@ ConnectionFSM.prototype.state_idle = function (S) {
 	if (this.cf_closeAfter === true) {
 		this.cf_closeAfter = false;
 		this.cf_lastError = undefined;
-		S.gotoState('closed');
 		S.on(this, 'closeAsserted', function () { });
+		S.gotoState('closed');
 		return;
 	}
 
diff --git a/package.json b/package.json
index dde76ba..f9dad64 100644
--- a/package.json
+++ b/package.json
@@ -1,6 +1,6 @@
 {
   "name": "cueball",
-  "version": "1.1.0",
+  "version": "1.1.1",
   "description": "",
   "main": "lib/index.js",
   "dependencies": {
